EasyPost/ClientConfiguration.cs

```diff
@@ -120,4 +120,37 @@ internal void SetUp()
 #pragma warning disable CA1307
     public override int GetHashCode() => ApiKey.GetHashCode() ^ ApiBase.GetHashCode() ^ Timeout.GetHashCode();
 #pragma warning restore CA1307
+
+    private bool _isDisposed;
+
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (_isDisposed) return;
+
+        if (disposing)
+        {
+            // Dispose managed state (managed objects).
+
+            // dispose of the prepared HTTP client
+            PreparedHttpClient?.Dispose();
+
+            // dispose of the user-provided HTTP client
+            CustomHttpClient?.Dispose();
+        }
+
+        // Free native resources (unmanaged objects) and override a finalizer below.
+        _isDisposed = true;
+    }
+
+    ~ClientConfiguration()
```

Comment:
what does `~` do here?

Reply:
This is a deconstructor, the opposite of a constructor. It runs when an object is destroyed in memory.